src/pages/iv-categories/forms/CategoryForm.js

```diff
@@ -0,0 +1,512 @@
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      recordId: null,
+      caRef: '',
+      name: '',
+      parentId: '',
+      lotCategoryId: '',
+      nraId: '',
+      spfId: '',
+      msId: '',
+      isMetal: false,
+      procurementMethod: '',
+      valuationMethod: '',
+      metalPurity: '',
+      metalId: '',
+      priceType: '',
+      taxId: '',
+      applyVAT: false,
+      allowNegativeQty: false,
+      isInactive: false
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      metalId: yup
+        .string()
+        .nullable()
+        .test('', function (value) {
+          const { isMetal } = this.parent
+          if (isMetal) {
+            return !!value
+          }
+
+          return true
+        }),
+      taxId: yup
+        .string()
+        .nullable()
+        .test('', function (value) {
+          const { applyVAT } = this.parent
+          if (applyVAT) {
+            return !!value
+          }
+
+          return true
+        }),
+      spfId: yup
+        .string()
+        .nullable()
+        .test('', function (value) {
+          const { trackBy } = this.parent
+          if (trackBy === 1 || trackBy === '1') {
+            return !!value
+          }
+
+          return true
+        }),
+      lotCategoryId: yup
+        .string()
+        .nullable()
+        .test('', function (value) {
+          const { trackBy } = this.parent
+          if (trackBy === 2 || trackBy === '2') {
+            return !!value
+          }
+
+          return true
+        }),
+      caRef: yup.string().required(),
+      name: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: InventoryRepository.Category.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!obj.recordId) {
+          toast.success(platformLabels.Added)
+
+          formik.setFieldValue('recordId', response.recordId)
+          setStore(prevStore => ({
+            ...prevStore,
+            recordId: response.recordId,
+            ref: formik.values.caRef,
+            name: formik.values.name
+          }))
+        } else {
+          toast.success(platformLabels.Edited)
+        }
+        invalidate()
+      } catch (error) {}
+    }
+  })
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: InventoryRepository.Category.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          setStore(prevStore => ({
+            ...prevStore,
+            ref: res.record.caRef,
+            name: res.record.name
+          }))
+
+          formik.setValues(res.record)
+        }
+      } catch (error) {}
+    })()
+  }, [])
+
+  useEffect(() => {
```

Comment:
Using onChange directly on the checkbox is sufficient to handle formik values updates, so there's no need for useEffect in this case.

example 
 ```
<Checkbox
                        name='applyVAT'
                        maxAccess={maxAccess}
                        checked={formik.values?.applyVAT}
                        onChange={e => {
                          formik.setFieldValue('applyVAT', e.target.checked)
                          if (e.target.checked == false) {
                            formik.setFieldValue('taxId', '')
                          }
                        }}
                      />
```